stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/util/HexToDecimal.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.ballerinalang.nativeimpl.util;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BInteger;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+
+/**
+ * Native function ballerina.util:hexToDecimal.
+ *
+ * @since 0.96.0
+ */
+@BallerinaFunction(
+        packageName = "ballerina.util",
+        functionName = "hexToDecimal",
+        args = {@Argument(name = "hexValue", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.INT)},
+        isPublic = true
+)
+public class HexToDecimal extends AbstractNativeFunction {
+
+    @Override
+    public BValue[] execute(Context context) {
+        String hexValue = getStringArgument(context, 0).substring(2);
+        return getBValues(new BInteger(Integer.parseInt(hexValue, 16)));
```

Comment:
The native function definition says that it returns an error as well. However, here, only the converted value is returned. If an exception is thrown during parsing, it will cause an error to be thrown in Ballerina, not returned.